R/recode-with-table.R

```diff
@@ -0,0 +1,1033 @@
+      rows_being_checked <-
+        rec_variables_to_process[rec_variables_to_process[[pkg.env$columns.Variables]] == variable_being_checked,]
+      rec_variables_to_process <-
+        rec_variables_to_process[!rec_variables_to_process[[pkg.env$columns.Variables]] == variable_being_checked,]
+      first_row <- rows_being_checked[1,]
+      # Check for varialbe existance in data
+      data_variable_being_checked <-
+        get_data_variable_name(
+          data_name = data_name,
+          row_being_checked = first_row,
+          variable_being_checked = variable_being_checked,
+          data = data
+        )
+      if (is.null(data[[data_variable_being_checked]])) {
+        warning(
+          paste(
+            "Data",
+            data_name,
+            "does not contain the variable",
+            data_variable_being_checked
+          )
+        )
+      } else {
+        # Check for From column duplicates
+        all_from_values_for_variable <-
+          rows_being_checked[[pkg.env$columns.From]]
+        if (length(unique(all_from_values_for_variable)) != length(all_from_values_for_variable)) {
+          for (single_from in all_from_values_for_variable) {
+            # Check if value is repeated more then once
+            if (sum(all_from_values_for_variable == single_from) > 1) {
+              stop(
+                paste(
+                  single_from,
+                  "was detected more then once in",
+                  variable_being_checked,
+                  "please make sure only one from value is being recoded"
+                )
+              )
+            }
+          }
+        }
+
+        # Set factor for all recode values
+        label_list[[variable_being_checked]] <-
+          create_label_list_element(rows_being_checked)
+        else_value <-
+          as.character(rows_being_checked[rows_being_checked[[pkg.env$columns.From]] == "else",
+                                          pkg.env$columns.CatValue])
+        if (length(else_value) == 1 &&
+            !is_equal(else_value, "character(0)")) {
+          else_value <-
+            recode_variable_NA_formating(else_value, label_list[[variable_being_checked]]$type)
+          if (is_equal(else_value, "copy")) {
+            recoded_data[variable_being_checked] <-
+              data[data_variable_being_checked]
+          } else {
+            recoded_data[variable_being_checked] <- else_value
+          }
+          # Catch multiple else rows
+        } else if (length(else_value) > 1) {
+          stop(
+            paste(
+              variable_being_checked,
+              " contains",
+              length(else_value),
+              "rows of else only one else value is allowed"
+            )
+          )
+        }
+        else {
+          recoded_data[variable_being_checked] <- else_default
+        }
+        else_rows <- nrow(recoded_data)
+        rows_being_checked <-
```

Comment:
I dont understand what you mean by this can you elaborate

Reply:
Mutating variables can lead to confusion because the programmer can forgot that the value of the variable has changed down the line. If possible always create new variables, like for this one I would create a new variables called `else_rows_being_checked`.

Reply:
But this removes all references of else from rows_being_checked. hence the else_rows_being_checked does not work here. Its cleaned rows_being_checked therefore i dont believe it warns variable name change i will add a comment describing this

Reply:
Yeah, I made a mistake, did not see the !. In that case you can rename it to `cleaned_rows_being_checked` or `non_else_rows_being_checked`.